src/ServiceControl.Transports.ASQ/ASQTransportCustomization.cs

```diff
@@ -23,6 +24,10 @@ static void ConfigureTransport(TransportExtensions<AzureStorageQueueTransport> t
             transport.SanitizeQueueNamesWith(BackwardsCompatibleQueueNameSanitizer.Sanitize);
             transport.Transactions(TransportTransactionMode.ReceiveOnly);
             transport.ConnectionString(transportSettings.ConnectionString);
+
+            transport.MessageInvisibleTime(TimeSpan.FromMinutes(5));
+            transport.BatchSize(1);
```

Comment:
Batch size of one?

Reply:
I'm not sure if it's applicable with ASQ, but we have had support cases (most recently with H&M) where they had to set the batch size for ASB to 1 in order to prevent Lock Lost exceptions. Daniel suggested it has something to do with prefetching and a bunch of other words I didn't understand.

I don't know what the correct values are here 😞

Reply:
I guess I can run a test with batch size set to 1 to verify the impact

Reply:
I don't understand setting this to `1`. In ASQ a message could reappear (as there are no locks) but what would take more than the configured 5min of  the  timeout invisibility?

Reply:
As the issue description says, it is expected for a handler to take several seconds, in that case it's a good practice to dial down the batch size.

@WilliamBZA correct values do not exist, for optimal performance it needs testing and adjustments to all the variables

Reply:
So there is no danger of messages going to the DLQ?

Reply:
ASQ has no DLQ. Messages will exhaust all the retries and will be sent to the Error queue.

Reply:
After talking with @SeanFeldman, larger batch sizes lower the number of requests to Azure Storage, which is a key billing metric.

The concern with a large batch size is timeouts. If the last message in the batch isn't processed before the timeout, another thread can pick it up.

Based on @SzymonPobiega's early testing of V3 I think default batch size is probably fine, and maybe bumping up the default timeout to 60 seconds would give us plenty of breathing room for those systems that are at the slower end of the spectrum, while allowing "abandoned" messages to reappear in the queue before or shortly after ServiceControl has restarted.

Reply:
I can always retest. Our goals here are, in this order:
 - Minimize the chances of moving messages to the error queue (because SC has no good support for moving them back)
 - Have acceptable throughput at least as high as the max processing rate we've seen in Azure (160 msg/s)
 - Minimize the cost